server/controllers/userController.ts

```diff
@@ -51,10 +52,40 @@ async function logout(req, res, next) {
 	res.sendStatus(200);
 }
 
+async function getUserData(req, res, next) {
+	const refreshToken = req.cookies.refreshToken;
+	const username = await userManager.findRefreshToken(refreshToken);
+	const data = await userManager.getUserData(username);
+	res.send(data);
+}
+
+async function changeUserIcon(req, res, next) {
+	const refreshToken = req.cookies.refreshToken;
+	const username = await userManager.findRefreshToken(refreshToken);
```

Comment:
I would highly recommend looking closer at these two lines which you use everywhere. Maybe, you can figure out how to make it nicer.